### PR TITLE
Use safe area layout to place ad in iOS 11

### PR DESCRIPTION
**Pull request by tlarsen7572**
This fixes ad placement for the iPhone X.  iOS code was updated per the [AdMob guidelines announced on October 24](https://support.google.com/admob/answer/7553599?ctx=ui&hl=en)

### CODE REVIEW

**Comment by Hixie** [comment]
cc one of @cbracken @HansMuller @collinjackson for review

File: packages/firebase_admob/ios/Classes/MobileAd.m [hunk 1]
```diff
@@ -218,11 +218,27 @@ - (void)show {
...
+
+#if __IPHONE_OS_VERSION_MAX_ALLOWED >= 110000
+  if (@available(ios 11.0, *)) {
+    UILayoutGuide *guide = screen.safeAreaLayoutGuide;
+    [NSLayoutConstraint activateConstraints:@[
+      [_banner.centerXAnchor constraintEqualToAnchor:guide.centerXAnchor],
+      [_banner.bottomAnchor constraintEqualToAnchor:guide.bottomAnchor]
+    ]];
+  } else {
+    CGFloat x = screen.frame.size.width / 2 - _banner.frame.size.width / 2;
```
Comment (HansMuller) [inline packages/firebase_admob/ios/Classes/MobileAd.m]: Please factor this code into a method that's called here and below the `#else`.

File: packages/firebase_admob/ios/Classes/MobileAd.m [hunk 2]
```diff
@@ -218,11 +218,27 @@ - (void)show {
   }
   if (_status != LOADED) return;
 
+  _banner.translatesAutoresizingMaskIntoConstraints = NO;
   UIView *screen = [FLTMobileAd rootViewController].view;
+  [screen addSubview:_banner];
+
+#if __IPHONE_OS_VERSION_MAX_ALLOWED >= 110000
```
Comment (cbracken) [inline packages/firebase_admob/ios/Classes/MobileAd.m]: Prefer: `#if defined(__IPHONE_11_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0)`

**Comment by HansMuller** [comment]
The changes look good, it looks like just a minor formatting tweak is needed.

Once Travis approves I'll update the version number, land the PR, and publish a new version of the plugin.

**Comment by tlarsen7572** [comment]
Thanks @HansMuller and @cbracken 

The Travis build failed with the Ruby test; it looks like it couldn't connect to Google to get the Dart SDK.  Is there a way to re-run that job?  I assume that is not an issue with the PR.

**Comment by HansMuller** [comment]
I restarted the analyzer job; all appears to be well.

I have to leave just now; will land this change first thing tomorrow.

**Comment by tlarsen7572** [comment]
Thanks @HansMuller!  Have a good night

**Comment by tlarsen7572** [comment]
Added CLA

**Comment by HansMuller** [comment]
Thanks for the CLA. I'll land this as soon as travis finishes.

**Comment by HansMuller** [comment]
The new version (0.2.1) of the plugin has been published: https://pub.dartlang.org/packages/firebase_admob

**Pull request merged**